From 61f7f0a959c7739ee322cb1809944cbc5b70463f Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Sat, 28 Dec 2013 21:06:18 +0100
Subject: [PATCH 13/16] BUG/MINOR: stream-int: do not clear the owner upon
 unregister

Since the applet rework and the removal of the inter-task applets,
we must not clear the stream-interface's owner task anymore otherwise
we risk a crash when maintaining keep-alive with an applet. This is
not possible right now so there is no impact yet, but this bug is not
easy to track down. No backport is needed.
---
 src/stream_interface.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/stream_interface.c b/src/stream_interface.c
index bf0a400..9aa1907 100644
--- a/src/stream_interface.c
+++ b/src/stream_interface.c
@@ -379,7 +379,6 @@ struct appctx *stream_int_register_handler(struct stream_interface *si, struct s
 void stream_int_unregister_handler(struct stream_interface *si)
 {
 	si_detach(si);
-	si->owner = NULL;
 }
 
 /* This callback is used to send a valid PROXY protocol line to a socket being
-- 
1.8.1.5

